File: neoforge/src/main/java/com/stal111/forbidden_arcanus/data/recipes/SpecialRecipesProvider.java
package com.stal111.forbidden_arcanus.data.recipes;

import com.stal111.forbidden_arcanus.common.item.crafting.CombineAurealTankRecipe;
import net.minecraft.core.HolderLookup;
import net.minecraft.data.recipes.SpecialRecipeBuilder;
import net.valhelsia.valhelsia_core.datagen.recipes.RecipeSubProvider;
import net.valhelsia.valhelsia_core.datagen.recipes.ValhelsiaRecipeProvider;

/**
 * @author stal111
 * @since 24.09.2023
 */
public class SpecialRecipesProvider extends RecipeSubProvider {

    public SpecialRecipesProvider(ValhelsiaRecipeProvider provider) {
        super(provider);
    }

    @Override
    protected void registerRecipes(HolderLookup.Provider lookupProvider) {
        SpecialRecipeBuilder.special(CombineAurealTankRecipe::new).save(this.getRecipeOutput(), "combine_aureal_tank");
    }
}

File: datagen/src/main/kotlin/com/stal111/forbidden_arcanus/datagen/recipes/SpecialRecipesProvider.kt
package com.stal111.forbidden_arcanus.datagen.recipes

import com.stal111.forbidden_arcanus.common.item.crafting.CombineAurealTankRecipe
import net.minecraft.core.HolderLookup
import net.minecraft.data.recipes.RecipeOutput
import net.minecraft.data.recipes.SpecialRecipeBuilder
import net.valhelsia.dataforge.recipe.RecipeSubProvider

class SpecialRecipesProvider(
    provider: HolderLookup.Provider,
    recipeOutput: RecipeOutput
) : RecipeSubProvider(provider, recipeOutput) {
    override fun buildRecipes() {
        SpecialRecipeBuilder.special { CombineAurealTankRecipe() }.save(this.recipeOutput, "combine_aureal_tank")
    }
}
